Content.Client/Construction/ConstructionSystem.cs

```diff
@@ -217,6 +216,27 @@ public void SpawnGhost(ConstructionPrototype prototype, EntityCoordinates loc, D
             return true;
         }
 
+        private bool CheckConstructionConditions(ConstructionPrototype prototype, EntityCoordinates loc, Direction dir,
+            EntityUid user)
+        {
+            foreach (var condition in prototype.Conditions)
+            {
+                if (!condition.Condition(user, loc, dir))
+                {
+                    var message = condition.GenerateGuideEntry()?.Localization;
+                    if (message != null)
+                    {
+                        // Show the reason to the user:
+                        _popupSystem.PopupCoordinates(Loc.GetString(message), loc);
+                    }
```

Comment:
I don't think these methods should be showing popups, they should just be pure and checking conditions, at best add a bool on whether to display the popup (also doesn't this spam them due to prediction?)

Reply:
I think this specific popup is appearing during a UI interaction while placing a construction ghost which is pure clientside. Perhaps I should add a mechanism for displaying popups outside my new condition (so that all conditions benefit from it)